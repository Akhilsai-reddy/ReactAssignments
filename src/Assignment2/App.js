import React, { Component } from 'react'
import AllProductsPage from './AllProductsPage'
export default class App extends Component {

  render() {
    return (
      <div>
     <AllProductsPage/>
      </div>
    )
  }
}
