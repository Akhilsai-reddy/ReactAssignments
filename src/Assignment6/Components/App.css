nav{
    background-color: aliceblue;
    padding: 16px 32px;
}
nav a{
    margin-right: 16px;
    color: blue;
}
nav a.active{
    color: rgb(245, 18, 18);
}