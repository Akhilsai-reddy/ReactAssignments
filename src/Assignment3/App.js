import React, { Component } from 'react'
import AllProductsPage from './AllProductsPage'
import { BrowserRouter,Switch,Route } from 'react-router-dom'
import Navbar from './Navbar'
import About from './About'
import './App.css'
export default class App extends Component {

  render() {
    return (
      <div>
        <BrowserRouter>
        <Navbar/>
        <Switch>
          <Route path='/' element={<About/>}/>
          <Route path='/Products' element={<AllProductsPage/>}/>
        </Switch>
        </BrowserRouter>
     
      </div>
    )
  }
}
