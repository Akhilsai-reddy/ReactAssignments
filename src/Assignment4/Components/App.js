import React, { Component } from 'react'
import AllProductsPage from './AllProductsPage'
import { BrowserRouter,Switch,Route } from 'react-router-dom'
import Navbar from './Navbar'
import About from './About'
import './App.css'
import ConetxtApi from './ConetxtApi'
import AddProduct from './AddProduct'
import ProdDetails from './ProdDetails'
export default class App extends Component {

  render() {
    return (
      <ConetxtApi>
        <BrowserRouter>
        <Navbar/>
        <Switch>
          <Route exact path='/' component={About}/>
          <Route exact path='/AddProduct' component={AddProduct}/>
          <Route exact path='/Products' render={()=><AllProductsPage/>}/>
          <Route exact path='/Products/:Product_Name' component={ProdDetails}/>
        </Switch>
        </BrowserRouter>
      </ConetxtApi>
    )
  }
}
