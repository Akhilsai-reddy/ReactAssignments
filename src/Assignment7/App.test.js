import App from "./App";
import { mount } from "enzyme";

describe('Rendering App',()=>{
    it('testing App',()=>{
        let wrapper=mount(<App/>)
        expect(wrapper).toMatchSnapshot();
    })
})