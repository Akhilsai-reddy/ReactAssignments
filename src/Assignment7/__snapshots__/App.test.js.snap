// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering App testing App 1`] = `
<App>
  <div>
    <AllProductsPage>
      <div>
        <h1>
          Products List
        </h1>
        <ProductList
          Products={
            Array [
              Object {
                "ID": 1,
                "Price": "$200",
                "Product_Name": "MOTO",
                "Quantity": 3,
              },
              Object {
                "ID": 2,
                "Price": "$400",
                "Product_Name": "REDMI",
                "Quantity": 3,
              },
              Object {
                "ID": 3,
                "Price": "$360",
                "Product_Name": "REALME",
                "Quantity": 3,
              },
            ]
          }
        >
          <table
            border={2}
          >
            <thead>
              <tr>
                <th>
                  ID
                </th>
                <th>
                  PRODUCT NAME
                </th>
                <th>
                  QUANTITY
                </th>
                <th>
                  PRICE
                </th>
              </tr>
            </thead>
            <tbody>
              <Product
                ID={1}
                Name="MOTO"
                Price="$200"
                Quantity={3}
                key="0"
              >
                <tr>
                  <td>
                    1
                  </td>
                  <td>
                    MOTO
                  </td>
                  <td>
                    3
                  </td>
                  <td>
                    $200
                  </td>
                </tr>
              </Product>
              <Product
                ID={2}
                Name="REDMI"
                Price="$400"
                Quantity={3}
                key="1"
              >
                <tr>
                  <td>
                    2
                  </td>
                  <td>
                    REDMI
                  </td>
                  <td>
                    3
                  </td>
                  <td>
                    $400
                  </td>
                </tr>
              </Product>
              <Product
                ID={3}
                Name="REALME"
                Price="$360"
                Quantity={3}
                key="2"
              >
                <tr>
                  <td>
                    3
                  </td>
                  <td>
                    REALME
                  </td>
                  <td>
                    3
                  </td>
                  <td>
                    $360
                  </td>
                </tr>
              </Product>
            </tbody>
          </table>
        </ProductList>
      </div>
    </AllProductsPage>
  </div>
</App>
`;
